test/test_gpmsa/scalar_pdf_small.C

```diff
@@ -0,0 +1,586 @@
+  }
+
+  // Experiment observation error (also needs to be scaled by
+  // simulation variance)
+  for (unsigned int i = 0; i < numExperiments; i++)
+    (*experimentMat)(i, i) = 1.0;
+
+  // Read in data and store the standard deviation of the simulation
+  // data (ignored for now).
+  const char * test_srcdir = std::getenv("srcdir");
+
+  std::string simInputFileName = "test_gpmsa/sim_scalar_small.dat";
+  std::string expInputFileName = "test_gpmsa/y_exp_scalar_small.txt";
+  std::string solInputFileName = "test_gpmsa/regression_solution_pdf_small.txt";
+
+  if (test_srcdir) {
+    simInputFileName = test_srcdir + ('/' + simInputFileName);
+    expInputFileName = test_srcdir + ('/' + expInputFileName);
+    solInputFileName = test_srcdir + ('/' + solInputFileName);
+  }
+
+  readData(simInputFileName,
+           expInputFileName,
+           simulationScenarios,
+           paramVecs,
+           outputVecs,
+           experimentScenarios,
+           experimentVecs,
+           *experimentMat);
+
+  // Add simulation and experimental data
+  gpmsaFactory.addSimulations(simulationScenarios, paramVecs, outputVecs);
+  gpmsaFactory.addExperiments(experimentScenarios, experimentVecs, experimentMat);
+
+  QUESO::GenericVectorRV<> postRv( "post_",
+      gpmsaFactory.prior().imageSet().vectorSpace());
+
+  QUESO::GslVector point(
+      gpmsaFactory.prior().imageSet().vectorSpace().zeroVector());
+
+  unsigned int num_lines = 10000;
+  std::vector<double> expected_log_likelihoods(num_lines);
+  std::vector<double> expected_log_priors(num_lines);
+  std::vector<double> expected_log_posteriors(num_lines);
+  std::vector<double> computed_log_likelihoods(num_lines);
+  std::vector<double> computed_log_priors(num_lines);
+  std::vector<double> computed_log_posteriors(num_lines);
+
+  // File containing x, theta, y (vertical concatenation of lhs.txt, y_mod.txt)
+  std::ifstream solution_data;
+  open_data_file(solInputFileName, solution_data);
+  for (unsigned int i = 0; i < num_lines; i++) {
+    for (unsigned int j = 0; j < point.sizeLocal(); ++j)
+      solution_data >> point[j];
+
+    double log_likelihood;
+    solution_data >> log_likelihood;
+    expected_log_likelihoods[i] = log_likelihood;
+
+    double log_prior;
+    solution_data >> log_prior;
+    expected_log_priors[i] = log_prior;
+
+    double log_posterior;
+    solution_data >> log_posterior;
+    expected_log_posteriors[i] = log_posterior;
+
+    double computed_log_prior;
+    computed_log_prior = (gpmsaFactory.prior().pdf().lnValue(point));
+    computed_log_priors[i] = computed_log_prior;
+
+    double computed_log_likelihood;
+    computed_log_likelihood = (gpmsaFactory.getGPMSAEmulator().lnValue(point, NULL, NULL, NULL, NULL));
+    computed_log_likelihoods[i] = computed_log_likelihood;
+  }
+  solution_data.close();
+
+  double initial_diff_prior = expected_log_priors[0] - computed_log_priors[0];
```

Comment:
This is because our log priors are correct only up to a constant, but we expect it to be the *same* constant at every evaluations?  We should have a comment about that.